tests/plt/plot_test.py

```diff
@@ -0,0 +1,168 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+import scipp as sc
+from scipp.plt import Plot, Figure, widgets, Node, node
+from ..factory import make_dense_data_array, make_dense_dataset
+import matplotlib
+import ipywidgets as ipw
+
+matplotlib.use('Agg')
+
+
+def test_plot_single_1d_line():
+    da = make_dense_data_array(ndim=1)
+    n = Node(da)
+    fig = Figure(n)
+    fig.render()
+
+
+def test_plot_two_1d_lines():
+    ds = make_dense_dataset(ndim=1)
+    a = Node(ds['a'])
+    b = Node(ds['b'])
+    fig = Figure(a, b)
+    fig.render()
+
+
+def test_plot_difference_of_two_1d_lines():
+    ds = make_dense_dataset(ndim=1)
+    a = Node(ds['a'])
+    b = Node(ds['b'])
+
+    @node
+    def diff(x, y):
+        return x - y
+
+    c = diff(a, b)
+    fig = Figure(a, b, c)
+    fig.render()
+
+
+def test_plot_2d_image():
+    da = make_dense_data_array(ndim=2)
+    a = Node(da)
+    fig = Figure(a)
+    fig.render()
+
+
+def test_plot_2d_image_smoothing_slider():
+    da = make_dense_data_array(ndim=2)
+    a = Node(da)
+
+    sl = ipw.IntSlider(min=1, max=10)
+    sigma_node = widgets.WidgetNode(sl)
+
+    from scipy.ndimage import gaussian_filter
+
+    @node
+    def smooth(da, sigma):
+        out = da.copy()
+        out.values = gaussian_filter(da.values, sigma=sigma)
+        return out
+
+    smooth_node = smooth(a, sigma=sigma_node)
+    fig = Figure(smooth_node)
+    Plot([fig, sl])
+    fig.render()
+    sl.value = 5
+
+
+def test_plot_2d_image_with_masks():
+    da = make_dense_data_array(ndim=2)
+    da.masks['m1'] = da.data < sc.scalar(0.0, unit='counts')
+    da.masks['m2'] = da.coords['xx'] > sc.scalar(30., unit='m')
+
+    a = Node(da)
+
+    widget = widgets.MaskWidget(da.masks)
+    w = widgets.WidgetNode(widget)
+
+    masks_node = widgets.hide_masks(a, w)
+    fig = Figure(masks_node)
+    Plot([fig, widget])
+    fig.render()
+    widget._all_masks_button.value = False
+
+
+def test_plot_two_1d_lines_with_masks():
+    ds = make_dense_dataset()
+    ds['a'].masks['m1'] = ds['a'].coords['xx'] > sc.scalar(40.0, unit='m')
+    ds['a'].masks['m2'] = ds['a'].data < ds['b'].data
+    ds['b'].masks['m1'] = ds['b'].coords['xx'] < sc.scalar(5.0, unit='m')
+
+    a = Node(ds['a'])
+    b = Node(ds['b'])
+
+    widget_a = widgets.MaskWidget(ds['a'].masks)
+    widget_b = widgets.MaskWidget(ds['b'].masks)
+    w_a = widgets.WidgetNode(widget_a)
+    w_b = widgets.WidgetNode(widget_b)
+
+    node_masks_a = widgets.hide_masks(a, w_a)
+    node_masks_b = widgets.hide_masks(b, w_b)
```

Comment:
I did not get this initially. Why do we need two things that are mask-related (widget and its node, `hide_masks`)?

I think I understand now, one provides the checkbox values, the other applies this mask-hiding to the data array.

Maybe my confusing came from the checkbox widget not actually being related to masks? Isn't it just a list of checkboxes? Then we could rename it and use it more generally?

I suppose instead of a `WidgetNode` I could pass a different node to `hide_masks`, e.g., to manually toggle masks, or some other GUI element (just asking conceptually speaking)?

Reply:
Yes, the `WidgetNode` is a node in the graph that provides the input to the node that actually hides the masks `node_masks_a/b`.

You are right that it is just a list of checkboxes. Maybe I can just make a generic checkbox widget.

Reply:
I suppose many of the typical widgets could be useful. For example a dropdown list to select which entry of a dataset should be plotted, etc. But we can keep all that for later, so long as we ensure that the mechanism is flexibly enough to use generic widgets.